mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/host/docker/CliJarDomainsAndUsersAdder.java

```diff
@@ -0,0 +1,57 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mpt.imapmailbox.external.james.host.docker;
+
+import org.apache.james.mpt.imapmailbox.external.james.ProvisioningException;
+import org.apache.james.mpt.imapmailbox.external.james.host.DomainsAndUsersAdder;
+import org.apache.james.util.docker.DockerGenericContainer;
+import org.testcontainers.containers.Container;
+import org.testcontainers.shaded.org.apache.commons.lang.StringUtils;
+
+import com.google.common.collect.ImmutableList;
+
+public class CliJarDomainsAndUsersAdder implements DomainsAndUsersAdder {
```

Comment:
I see it's a little bit verbose with this class and it's parent. Why don't use the `ServerCmd.executeAndOutputToStream()` ? we just need to expose the 9999 port from containers to the host, then 
will be able to use that tool.

Do it we can save plenty of classes.

Reply:
I liked your idea.
I just tried to implement it, but was not able to make it works. 
Because the jvm inside the docker image must be launched with parameters to allow remote JMX connections which is not the case.